### DISCUSSION TITLE
lightweight deletes not compatible with projections?

### DISCUSSION THREAD

**Discussion by beda42** [ClickHouse/ClickHouse#49299] [Q&A] [answered]
I wanted to test the new lightweight deletes but for my table it returned `DB::Exception: DELETE query is not supported for table ...`. After some experimentation, I found that this happens if the table has a projection. I was not able to find anything about this in the docs and only one commit from about 6 months ago.

So my question is:

* is this a known limitation which is just lacking documentation?
* is this limitation temporary or does it come from the nature of projections?
* if it is temporary, any idea when support for tables with projections will be added?

Thanks for any hints.

**Comment by UnamedRus** [comment, marked as answer]
> is this a known limitation which is just lacking documentation?

I guess, there is no explicit issue for that and no mention in docs (this should be fixed), but it's known.

https://github.com/ClickHouse/ClickHouse/pull/37893/files#diff-dc0e615fd2eb67c2b328ccd9290cc56ed10dce13d05d58c2270341f0c49b23b8R1194-R1200

> is this limitation temporary or does it come from the nature of projections?

It's a bit tricky to implement support for projections here, but i guess it will be done in future.

**Reply by beda42** [reply]
Thanks for the answer :+1: 

I guess I will have to wait to be able to use lightweight deletes then.
